pkg/http-tracer/http-trace.go

```diff
@@ -0,0 +1,202 @@
+/*
```

Comment:
Keep this package to be `handlers` instead of http-trace is too high level, also please bring a test case for this package. `handlers.TraceReqHandlerFunc` , we can move few other handlers into this package as well.  Eventually..